dynamic_listings/python/default/design_patterns/chain_of_responsibility.py

```diff
@@ -1,19 +1,22 @@
+from typing import Self
+
+
 class Handler:
     # This is the handler abstract/class interface that the sender connects to
-    next = None  # The next handler in the chain
+    next_handler: Self = None  # The next handler in the chain
 
-    def handle_request(self):
+    def handle_request(self) -> None:
         if condition:
             # In case I can handle this request
             return self.real_handler()
 
-        if self.next is not None:
-            self.next.handle_request()
+        if self.next_handler:
+            self.next_handler.handle_request()
```

Comment:
I think this is missing a return, or the chain will be broken after the first non-handling. Need to check the other sources too.